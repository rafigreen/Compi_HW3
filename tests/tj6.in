//Ambiguous ambiguous? ambiguous ambiguous. AMBIGUOUS!

override bool f(){int x = 1;}
override int f(){int x = 1;}

override int g(bool y){int x = 1;}
override bool g(int y){int x = 1;}

void main(){
    int a = g(f());
}