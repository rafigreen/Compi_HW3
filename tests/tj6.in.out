---end scope---
x INT 0
---end scope---
x INT 0
---end scope---
y BOOL -1
x INT 0
---end scope---
y INT -1
x INT 0
line 10: ambiguous call to overloaded function f
